From 9cd51a5d786aaae0d165aff51564986010c4c6aa Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Mon, 29 Jan 2018 11:48:23 +0800
Subject: [PATCH] usb: dwc3: rockchip: use msleep() for long delays

ulseep_range() uses hrtimers and provides no advantage over msleep()
for larger delays. Fix up the 20+ ms delays here passing the adjusted
"min" value to msleep().

Change-Id: Ideb6798d3a9465b64dee5aaf78a73054474dd017
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 drivers/usb/dwc3/dwc3-rockchip.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/dwc3-rockchip.c b/drivers/usb/dwc3/dwc3-rockchip.c
index f52d8540e5bb..c018b2007922 100644
--- a/drivers/usb/dwc3/dwc3-rockchip.c
+++ b/drivers/usb/dwc3/dwc3-rockchip.c
@@ -322,8 +322,8 @@ static ssize_t dwc3_rockchip_host_testmode_write(struct file *file,
 				    EXTCON_PROP_USB_TYPEC_POLARITY, property);
 		extcon_set_cable_state_(edev, EXTCON_USB_HOST, true);
 
-		/* Add a delay 1~1.5s to wait for XHCI HCD init */
-		usleep_range(1000000, 1500000);
+		/* Add a delay 1s to wait for XHCI HCD init */
+		msleep(1000);
 	}
 
 	dwc3_rockchip_set_test_mode(rockchip, testmode);
-- 
2.35.3

